test_runner/batch_others/test_wal_acceptor.py

```diff
@@ -14,13 +14,43 @@
 from dataclasses import dataclass, field
 from multiprocessing import Process, Value
 from pathlib import Path
-from fixtures.neon_fixtures import PgBin, Etcd, Postgres, RemoteStorageUsers, Safekeeper, NeonEnv, NeonEnvBuilder, PortDistributor, SafekeeperPort, neon_binpath, PgProtocol
+from fixtures.neon_fixtures import NeonPageserver, PgBin, Etcd, Postgres, RemoteStorageUsers, Safekeeper, NeonEnv, NeonEnvBuilder, PortDistributor, SafekeeperPort, neon_binpath, PgProtocol, wait_for_last_record_lsn, wait_for_upload
 from fixtures.utils import get_dir_size, lsn_to_hex, lsn_from_hex
 from fixtures.log_helper import log
 from typing import List, Optional, Any
 from uuid import uuid4
 
 
+def wait_lsn_force_checkpoint(tenant_id: str,
+                              timeline_id: str,
+                              pg: Postgres,
+                              ps: NeonPageserver,
+                              pageserver_conn_options={}):
+    lsn = lsn_from_hex(pg.safe_psql('SELECT pg_current_wal_flush_lsn()')[0][0])
+    log.info(f"pg_current_wal_flush_lsn is {lsn_to_hex(lsn)}, waiting for it on pageserver")
+
+    auth_token = None
+    if 'password' in pageserver_conn_options:
+        auth_token = pageserver_conn_options['password']
+
+    # wait for the pageserver to catch up
+    wait_for_last_record_lsn(ps.http_client(auth_token=auth_token),
+                             uuid.UUID(hex=tenant_id),
+                             uuid.UUID(hex=timeline_id),
+                             lsn)
+
+    # force checkpoint to advance remote_consistent_lsn
+    with closing(ps.connect(**pageserver_conn_options)) as psconn:
+        with psconn.cursor() as pscur:
+            pscur.execute(f"checkpoint {tenant_id} {timeline_id}")
+
+    # ensure that remote_consistent_lsn is advanced
+    wait_for_upload(ps.http_client(auth_token=auth_token),
+                    uuid.UUID(hex=tenant_id),
+                    uuid.UUID(hex=timeline_id),
+                    lsn)
+
```

Comment:
This isn't required for these tests, AFAICS

Reply:
Tests are waiting for things that depend on `remote_consistent_lsn` propagation and will work without `wait_for_upload`. I'd leave this check just to confirm that pageserver is actully moving `remote_consistent_lsn`.